queue/index.php

```diff
@@ -2,129 +2,139 @@
-					$USER = new User();
-					if ($USER->isLoggedIn() && !isset($_REQUEST['public'])) { // we enable logged in users to get to the public queue by having ?public in the URL
-				?>
-					<private-queue id="queueApp" class="wdn-band">
-						<script src="/dist/queue/queuePrivate.js"></script>
-					</private-queue>
-				<?php
-					} else {
-				?>
-					<public-queue id="queueApp" class="wdn-band">
-						<script src="/dist/queue/queuePublic.js"></script>
-					</public-queue>
-				<?php
-					}
-				?>
-				<!-- TemplateEndEditable -->
-			</div>
-		</main>
-		<footer id="footer" role="contentinfo" class="wdn-content-slide">
-			<div id="wdn_optional_footer" class="wdn-band wdn-footer-optional">
-				<div class="wdn-inner-wrapper">
-					<!-- TemplateBeginEditable name="optionalfooter" -->
-					<!-- TemplateEndEditable -->
-				</div>
-			</div>
-			<div id="wdn_local_footer" class="wdn-band wdn-footer-local">
-				<div class="wdn-inner-wrapper">
-					<!-- TemplateBeginEditable name="contactinfo" -->
-					<?php include "$root/sharedcode/localFooter.html"; ?>
-					<!-- TemplateEndEditable -->
-					<!-- TemplateBeginEditable name="leftcollinks" -->
-					<!-- TemplateEndEditable -->
-				</div>
-			</div>
-			<div id="wdn_global_footer" class="wdn-band wdn-footer-global">
-				<div class="wdn-inner-wrapper">
-					<?php wdnInclude("/wdn/templates_4.1/includes/globalfooter.html"); ?>
-				</div>
-			</div>
-		</footer>
-		<?php wdnInclude("/wdn/templates_4.1/includes/noscript.html"); ?>
+		<!-- TemplateEndEditable -->
+	</div>
+	<div class="dcf-main-content dcf-wrapper dcf-p-0">
+		<!-- TemplateBeginEditable name="maincontentarea" -->
+		<vita-queue id="queueApp" />
+		<!-- TemplateEndEditable -->
 	</div>
-	<?php wdnInclude("/wdn/templates_4.1/includes/body_scripts.html"); ?>
-	<?php require_once "$root/server/global_includes.php"; ?>
+</main>
+<footer class="dcf-footer" id="dcf-footer" role="contentinfo">
+	<!-- TemplateBeginEditable name="optionalfooter" -->
+	<!-- TemplateEndEditable -->
+		<?php wdnInclude("/wdn/templates_5.0/includes/global/footer-global-1.html"); ?>
+	<!-- TemplateBeginEditable name="contactinfo" -->
+		<?php include "$root/sharedcode/localFooter.html"; ?>
+	<!-- TemplateEndEditable -->
+		<?php wdnInclude("/wdn/templates_5.0/includes/global/footer-global-2.html"); ?>
+</footer>
+<?php wdnInclude("/wdn/templates_5.0/includes/global/noscript.html"); ?>
+<?php wdnInclude("/wdn/templates_5.0/includes/global/js-body.html"); ?>
+<!-- TemplateBeginEditable name="jsbody" -->
+<?php require_once "$root/server/global_includes.php"; ?>
+<?php
+	require_once "$root/server/user.class.php";
+	$USER = new User();
+	if ($USER->isLoggedIn() && !isset($_REQUEST['public'])) { // we enable logged in users to get to the public queue by having ?public in the URL
+?>
+	<script src="/dist/queue/queuePrivate.js"></script>
+<?php
+	} else {
+?>
+	<script src="/dist/queue/queuePublic.js"></script>
```

Comment:
I realize this all works... but how does this work? How does the `vita-queue` directive used above know to use the private queue or the public queue PHP code? We now have two directives with the same name (vitaQueue), how does Angular know which one to use?

Reply:
Ah I think I get it... this part here determines which angular directive. Brilliant